pysb/importers/bngl.py

```diff
@@ -0,0 +1,319 @@
+
+    def parse_monomers(self):
+        for m in self.x.iterfind(_ns('{0}ListOfMoleculeTypes/'
+                                     '{0}MoleculeType')):
+            mon_name = m.get('id')
+            sites = []
+            states = {}
+            for ctype in m.iterfind(_ns('{0}ListOfComponentTypes/'
+                                        '{0}ComponentType')):
+                c_name = ctype.get('id')
+                sites.append(c_name)
+                states_list = ctype.find(_ns('{}ListOfAllowedStates'))
+                if states_list is not None:
+                    states[c_name] = [s.get('id') for s in
+                                      states_list.iterfind(_ns(
+                                                           '{}AllowedState'))]
+            try:
+                self.monomer(mon_name, sites, states)
+            except Exception as e:
+                if str(e).startswith('Duplicate sites specified'):
+                    raise BnglImportError('Molecule %s has multiple '
+                                          'sites with the same name. '
+                                          'This is not supported in '
+                                          'PySB.' % mon_name)
+                else:
+                    raise BnglImportError(str(e))
+
+    def parse_parameters(self):
+        for p in self.x.iterfind(_ns('{0}ListOfParameters/{0}Parameter')):
+            p_name = p.get('id')
+            if p.get('type') == 'Constant':
+                p_value = p.get('value').replace('10^', '1e')
+                self.parameter(name=p_name, value=p_value)
+            elif p.get('type') == 'ConstantExpression':
+                self.expression(name=p_name,
+                                expr=self._eval_in_model_env(p.get('value')))
+            else:
+                raise BnglImportError('Parameter %s has unknown type: %s' %
+                                      (p_name, p.get('type')))
+
+    def parse_observables(self):
+        for o in self.x.iterfind(_ns('{0}ListOfObservables/{0}Observable')):
+            o_name = o.get('name')
+            cplx_pats = []
+            for mp in o.iterfind(_ns('{0}ListOfPatterns/{0}Pattern')):
+                match_once = mp.get('matchOnce')
+                match_once = 1 if match_once is not None \
+                    and match_once == "1" else 0
+                cplx_pats.append(ComplexPattern(self._parse_species(mp),
+                                                compartment=None,
+                                                match_once=match_once))
+            self.observable(o_name,
+                            ReactionPattern(cplx_pats),
+                            match=o.get('type').lower())
+
+    def parse_initials(self):
+        for i in self.x.iterfind(_ns('{0}ListOfSpecies/{0}Species')):
+            if i.get('Fixed') is not None and i.get('Fixed') == "1":
+                warnings.warn('Species %s is fixed, but will be treated as '
```

Comment:
Unimplemented functionality which changes model behaviour should probably error by default, yes. It'd be good to have a `force=True` option to build the model partially anyway, so that advanced users can fix it with manual edits.